include/swift/ClangImporter/ClangImporterRequests.h

```diff
@@ -342,6 +342,12 @@ SourceLoc extractNearestSourceLoc(CxxRecordSemanticsDescriptor desc);
 
 /// What pattern does this C++ API fit? Uses attributes such as
 /// import_owned and import_as_reference to determine the pattern.
+///
+/// Do not evaluate this request before importing has started. For example, it
```

Comment:
Is there a way we could assert this property or invoke the necessary logic during the request? This seems like an easy mistake to make again.

(This would be an enhancement—I would not have rejected the PR for this.)